tests/e2e/ctl_v3_member_test.go

```diff
@@ -309,3 +312,104 @@ func TestRemoveNonExistingMember(t *testing.T) {
 	err = epc.Restart(ctx)
 	assert.NoError(t, err)
 }
+
+func TestMemberReplace(t *testing.T) {
+    e2e.BeforeTest(t)
+    ctx := context.Background()
+
+    epc, err := e2e.NewEtcdProcessCluster(ctx, t)
+    assert.NoError(t, err)
+    defer epc.Close()
+
+	memberId := rand.Int() % len(epc.Procs)
+	member := epc.Procs[memberId]
+	var endpoints []string
+	for i := 1; i < len(epc.Procs); i++ {
+		endpoints = append(endpoints, epc.Procs[(memberId+i)%len(epc.Procs)].EndpointsGRPC()...)
+	}
+	cc, err := e2e.NewEtcdctl(epc.Cfg.Client, endpoints)
+	assert.NoError(t, err)
+
+	c := epc.Etcdctl()
+	memberID, found, err := getID(ctx, c, member.Config().Name)
+	assert.NoError(t, err)
+	assert.Equal(t, found, true, "Member not found")
+
+	// Need to wait health interval for cluster to accept member changes
+	time.Sleep(etcdserver.HealthInterval)
+	_, err = c.MemberRemove(ctx, memberID)
+	assert.NoError(t, err)
+	_, found, err = getID(ctx, c, member.Config().Name)
+	assert.NoError(t, err)
+	assert.Equal(t, found, false, "Expected member to be removed")
+
+	t.Logf("Removing member %s", member.Config().Name)
+	for member.IsRunning() {
+		member.Kill()
```

Comment:
Double checked the Kill here is not needed (mistake in the original test), the removed member should exit by themselves. You can just replace the whole block with `member.Wait(ctx)` to confirm that member exits.